File: backend/src/main/java/com/keyboardmarket/model/GroupBuy.java
package com.keyboardmarket.model;

import java.time.Instant;
import java.util.List;
import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

@Document(collection = "group-buys")
@Data
public class GroupBuy {

    @Id
    private String id;

    @Field("topic_id")
    private String topicId;

    private String name;
    private String type;
    private String status;
    private String designer;
    private String overview;
    private String poster;

    @Field("gb_start")
    private String gbStart;

    @Field("gb_end")
    private String gbEnd;

    @Field("estimated_fulfillment")
    private String estimatedFulfillment;

    @Field("base_price")
    private BasePrice basePrice;

    private List<Item> items;
    private List<Vendor> vendors;

    @Field("discord_url")
    private String discordUrl;

    @Field("source_url")
    private String sourceUrl;

    private List<String> images;

    @Field("scraped_at")
    private Instant scrapedAt;

    @Data
    public static class BasePrice {
        private Double amount;
        private String currency;
    }

    @Data
    public static class Item {
        private String name;
        private Double price;
        private String currency;
    }

    @Data
    public static class Vendor {
        private String region;
        private String name;
        private String url;
    }
}

File: frontend/src/types/groupBuy.ts
export interface BasePrice {
  amount: number;
  currency: string;
}

export interface Item {
  name: string;
  price: number;
  currency: string;
  imageUrl?: string;
}

export interface Vendor {
  region: string;
  name: string;
  url: string;
}

interface GroupBuyBase {
  name: string;
  type: string;
  status: string;
  designer: string;
  overview: string | null;
  gbStart: string | null;
  gbEnd: string | null;
  estimatedFulfillment: string | null;
  basePrice: BasePrice | null;
  items: Item[];
  vendors: Vendor[];
  discordUrl: string | null;
  images: string[];
}

export interface ApiGroupBuy extends GroupBuyBase {
  id: string;
  topicId: string;
  sourceUrl: string;
  featured?: boolean;
}

export interface AdminGroupBuy extends GroupBuyBase {
  id?: string;
  topicId?: string;
  sourceUrl: string | null;
  poster: string | null;
  excludedImages: string[];
  hidden: boolean;
  featured?: boolean;
}
